internal/compiler-interface/src/main/java/xsbti/AnalysisCallback.java

```diff
@@ -11,54 +11,171 @@
+     * @param classApi The extracted public class API.
+     */
     void api(File sourceFile, xsbti.api.ClassLike classApi);
+
+    /**
+     * Register the use of a <code>name</code> from a given source class name.
+     *
+     * @param className The source class name that uses <code>name</code>.
+     * @param name The source name used in <code>className</code>.
+     */
     void usedName(String className, String name);
-    /** Provides problems discovered during compilation.  These may be reported (logged) or unreported.
-     * Unreported problems are usually unreported because reporting was not enabled via a command line switch. */
-    void problem(String what, Position pos, String msg, Severity severity, boolean reported);
+
+    /**
+     * Register a compilation problem.
+     *
+     * This error may have already been logged or not. Unreported problems may
+     * happen because the reporting option was not enabled via command-line.
+     *
+     * @param what The headline of the error.
+     * @param pos At a given source position.
+     * @param msg The in-depth description of the error.
+     * @param severity The severity of the error reported.
+     * @param reported Flag that confirms whether this error was reported or not.
+     */
+    void problem(String what,
+                 Position pos,
+                 String msg,
+                 Severity severity,
+                 boolean reported);
+
     /**
-     * Determines whether method calls through this interface should be interpreted as serving
-     * name hashing algorithm needs in given compiler run.
+     * Determine whether method calls through this interface should be
+     * interpreted to use the name-hashing algorithm in the given compiler run.
      *
-     * In particular, it indicates whether member reference and inheritance dependencies should be
-     * extracted.
      *
-     * As the signature suggests, this method's implementation is meant to be side-effect free. It's added
-     * to AnalysisCallback because it indicates how other callback calls should be interpreted by both
-     * implementation of AnalysisCallback and it's clients.
+     * In particular, it indicates whether member references, inheritance
+     * dependencies and local inherited dependencies should be extracted.
      *
-     * NOTE: This method is an implementation detail and can be removed at any point without deprecation.
-     *       Do not depend on it, please.
+     * @see xsbti.api.DependencyContext for more information on the sort of
+     * dependency that can be registered.
+     *
+     * The implementation of this method is meant to be free of side-effects.
+     * It's added to {@link AnalysisCallback} because it indicates how other
+     * callback calls should be interpreted by its implementations and clients.
+     *
+     * This method is an implementation detail and can be removed at
+     * any point without any deprecation. Do not depend on it.
```

Comment:
(Also potentially off topic on this PR) Perhaps we should just remove it now. This is a boolean flag indicating it uses source-based name hashing as opposed to the older incremental compiler. Now we no longer have either of the algorithms, and we have new class-based name hashing.